aptos-move/block-executor/src/executor.rs

```diff
@@ -136,38 +136,60 @@ where
             .modified_keys(idx_to_execute)
             .map_or(HashMap::new(), |keys| keys.collect());
 
+        let mut prev_modified_aggregators = last_input_output
+            .aggregator_v2_keys(idx_to_execute)
+            .map_or(HashSet::new(), |keys| keys.collect());
+
         // For tracking whether the recent execution wrote outside of the previous write/delta set.
         let mut updates_outside = false;
-        let mut apply_updates = |output: &E::Output| {
-            // First, apply writes.
-            for (k, v) in output
-                .resource_write_set()
-                .into_iter()
-                .chain(output.aggregator_v1_write_set().into_iter())
-            {
-                if prev_modified_keys.remove(&k).is_none() {
-                    updates_outside = true;
+        let mut apply_updates =
+            |output: &E::Output| {
+                // First, apply writes.
+                for (k, v) in output
+                    .resource_write_set()
+                    .into_iter()
+                    .chain(output.aggregator_v1_write_set().into_iter())
+                {
+                    if prev_modified_keys.remove(&k).is_none() {
+                        updates_outside = true;
+                    }
+                    versioned_cache
+                        .data()
+                        .write(k, idx_to_execute, incarnation, v);
                 }
-                versioned_cache
-                    .data()
-                    .write(k, idx_to_execute, incarnation, v);
-            }
 
-            for (k, v) in output.module_write_set().into_iter() {
-                if prev_modified_keys.remove(&k).is_none() {
-                    updates_outside = true;
+                for (k, v) in output.module_write_set().into_iter() {
+                    if prev_modified_keys.remove(&k).is_none() {
+                        updates_outside = true;
+                    }
+                    versioned_cache.modules().write(k, idx_to_execute, v);
                 }
-                versioned_cache.modules().write(k, idx_to_execute, v);
-            }
 
-            // Then, apply deltas.
-            for (k, d) in output.aggregator_v1_delta_set().into_iter() {
-                if prev_modified_keys.remove(&k).is_none() {
-                    updates_outside = true;
+                // Then, apply deltas.
+                for (k, d) in output.aggregator_v1_delta_set().into_iter() {
+                    if prev_modified_keys.remove(&k).is_none() {
+                        updates_outside = true;
+                    }
+                    versioned_cache.data().add_delta(k, idx_to_execute, d);
+                }
+
+                for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                    if !prev_modified_aggregators.remove(&id) {
```

Comment:
Is this same as `prev_modified_aggregators.remove(&id).is_none()` as used in the above other for loops?

Reply:
prev_modified_aggregators is HashSet, prev_modified_keys is HashMap, and that's where the difference comes from